internal/controllers/machine/machine_controller.go

```diff
@@ -352,21 +352,30 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Clu
 
 			conditions.MarkTrue(m, clusterv1.DrainingSucceededCondition)
 			r.recorder.Eventf(m, corev1.EventTypeNormal, "SuccessfulDrainNode", "success draining Machine's node %q", m.Status.NodeRef.Name)
+		}
+
+		// After node draining is completed, and if isNodeVolumeDetachingAllowed returns True, make sure all
+		// volumes are detached before proceeding to delete the Node.
+		if r.isNodeVolumeDetachingAllowed(m) {
+			log.Info("Waiting for node volumes to be detached", "Node", klog.KRef("", m.Status.NodeRef.Name))
```

Comment:
This log message duplicates the one shown in line 374. Would be good to remove one of them. If we keep this one, we should add a second log entry when volumes are detached. Otherwise it makes debugging harder - caused a lot of confusion for us just now.
I can move this to an issue if necessary.

Reply:
Would be good if you can either move it to an issue or open a PR if you want to